project_resource_calendar/tests/test_calendar_event.py

```diff
@@ -0,0 +1,17 @@
+# © 2018 Savoir-faire Linux
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/LGPL).
+
+
+from odoo.addons.project_resource_calendar.tests.common import TestCalendarEventCommon
+
+
+class TestCalendarEvent(TestCalendarEventCommon):
+
+    def setUp(self):
+        super(TestCalendarEvent, self).setUp()
+        self.validate_room_name = 'Testing_Room_type_2'
+        self.validate_instrument_name = 'Testing_Instrument_type_2'
+
+    def test_010_onchange_room_id(self):
```

Comment:
To test on_change method you should validate the state before and after and validate that what you are changing actually changed ( the purpose of the on change room id method). Maybe I don’t get it, but what i’m see here, is that you are only asserting the name after create, not on_change

Reply:
you have to check the onchange_room_id method and to check the values before and after the execution of this method

Reply:
Change it